bchydro/api.py

```diff
@@ -73,20 +73,24 @@ def _parse_bchydroparam(self, soup: BeautifulSoup) -> str:
 
     def _validate_html_response(self, html) -> BeautifulSoup:
         soup = None
-        alert_errors = None
 
         try:
             soup = BeautifulSoup(html, features="html.parser")
-            alert_errors = soup.find(True, {"class": ["alert", "error"]})
             self._bchydroparam = self._parse_bchydroparam(soup)
         except:
             raise BCHydroInvalidHtmlException()
 
+        return soup
+
+    def _detect_alert_errors(self, soup: BeautifulSoup):
+        try:
+            alert_errors = soup.select('.alert.error:not(.hidden)')[0]
```

Comment:
This `[0]` throws an IndexError for me; I suggest moving it to L92 `alert_errors[0].text`

With that fixed, this works for my multi-address account, while `master` currently reproduces #6 :)

I do notice raising a BCHydroAlertDialogException() here triggers several retries, so the ultimate exception coming from test.py is `RetryError`, which seems like it'd somewhat obscure auth issues from bubbling up (eg. to HA) - I'm not sure what your use case was for adding auto retry, but allowing explicit (optional) calls of `.authenticate()`, while keeping your "(re)-authenticate if needed" in the refresh functions.